packages/core/admin/admin/src/content-manager/history/pages/History.tsx

```diff
@@ -90,6 +94,19 @@ const HistoryPage = () => {
     { refetchOnMountOrArgChange: true }
   );
 
+  /**
+   * When the page is first mounted, if there's already data in the cache, RTK has a fullfilled
+   * status for the first render, right before it triggers a new request. This means the code
+   * briefly reaches the part that redirects to the first history version (if none is set).
+   * But since that data is stale, that means auto-selecting a version that may not be the most
+   * recent. To avoid this, we identify through requestId if the query is stale despite the
+   * fullfilled status, and show the loader in that case.
+   * This means we essentially don't want cache. We always refetch when the page mounts, and
+   * we always show the loader until we have the most recent data. That's fine for this page.
+   */
+  const initialRequestId = React.useRef(versionsResponse.requestId);
+  const isStaleRequest = versionsResponse.requestId === initialRequestId.current;
```

Comment:
8 line comment for a 2 liner that fixes an annoying bug. I dig your style. Works for me, nice fix.